LEAF_Request_Portal/admin/templates/mod_form.tpl

```diff
@@ -83,181 +91,182 @@ function editProperties(isSubForm) {
-                        CSRFToken: '<!--{$CSRFToken}-->'},
-                    success: function(res) {
-                        categories[currCategoryID].description = $('#description').val();
-                    }
-                }));
+            else {
+                $('#container_workflowID').html('<span style="color: red">A workflow must be set up first</span>');
             }
+            dialog.indicateIdle();
+        },
+        cache: false
+    });
 
-            if(workflowChanged){
-                calls.push(
-                    $.ajax({
-                    type: 'POST',
-                    url: '../api/?a=formEditor/formWorkflow',
-                    data: {workflowID: $('#workflowID').val(),
-                    	categoryID: currCategoryID,
-                        CSRFToken: '<!--{$CSRFToken}-->'},
-                    success: function(res) {
-                        if(res == false) {
-                        	alert('Workflow cannot be set because this form has been merged into another form');
-                        }
-                        categories[currCategoryID].workflowID = $('#workflowID').val();
-                    }
-                }));
-            }
+    dialog.setSaveHandler(function() {
+        let calls = [];
+        let nameChanged = (categories[currCategoryID].categoryName || "") != $('#name').val();
+        let descriptionChanged  = (categories[currCategoryID].categoryDescription || "") != $('#description').val();
+        let workflowChanged  = (categories[currCategoryID].workflowID || "") != $('#workflowID').val();
+        let needToKnowChanged = (categories[currCategoryID].needToKnow || "") != $('#needToKnow').val();
+        let sortChanged = (categories[currCategoryID].sort || "") != $('#sort').val();
+        let visibleChanged = (categories[currCategoryID].visible || "") != $('#visible').val();
+        let typeChanged = (categories[currCategoryID].type || "") != $('#formType').val();
 
-            if(needToKnowChanged){
-                calls.push(
-                $.ajax({
-                    type: 'POST',
-                    url: '../api/?a=formEditor/formNeedToKnow',
-                    data: {needToKnow: $('#needToKnow').val(),
-                        categoryID: currCategoryID,
-                        CSRFToken: '<!--{$CSRFToken}-->'},
-                    success: function(res) {
-                        categories[currCategoryID].needToKnow = $('#needToKnow').val();
-                    }
-                }));
-            }
+        if(nameChanged){
+            calls.push($.ajax({
+                type: 'POST',
+                url: '../api/?a=formEditor/formName',
+                data: {name: $('#name').val(),
+                    categoryID: currCategoryID,
+                    CSRFToken: '<!--{$CSRFToken}-->'},
+                success: function(res) {
+                    categories[currCategoryID].name = $('#name').val();
+                }
+            }));
```

Comment:
This one definitely needs a fail-catch.  Data isn't being written that is expected to be written.